### PR TITLE
Slight tweaks to open it up to my usage (and maybe others)

### PR DESCRIPTION
**Pull request by adamjack**
Very basic, but just trying to return what others might find useful.

1) Allowed passing of the product/service string (so I can set it to my service publicly, even though it is your code.)
2) Allowed Serialization to an io.Writer not just string (better error handling here would be a nice addition, later.) 
3) Made an example RRule in the README 'cos I wasn't sure how to make that call (I assumed all sorts of KeyValue parameters.)

P.S. I'm not as familiar with pull requests as I'd like to be, so please forgive/correct any protocol mistakes.

### CODE REVIEW

File: calendar.go [hunk 1]
```diff
@@ -275,26 +275,35 @@ type Calendar struct {
 }
 
 func NewCalendar() *Calendar {
+	return NewCalendarFor("Arran Ubels")
```
Comment (arran4) [inline calendar.go:278]: I completely forgot about this. -- That shouldn't be there. Perhaps make it arran4?

File: README.md [hunk 2]
```diff
@@ -23,6 +23,7 @@ Creating:
   event.SetLocation("Address")
   event.SetDescription("Description")
   event.SetURL("https://URL/")
+  event.AddRrule(fmt.Sprintf("FREQ=YEARLY;BYMONTH=%d;BYMONTHDAY=%d", time.Now().Month(), time.Now().Day()))
```
Comment (arran4) [inline README.md:26]: I will have to figure out better way of communicating that info. That's a bit cumbersome. But good for now.

**Comment by arran4** [comment]
@adamjack Hey!

Thanks for the PR. PR's are almost always appreciated!

> P.S. I'm not as familiar with pull requests as I'd like to be, so please forgive/correct any protocol mistakes.

I wouldn't worry about that you have preformed better than average. 👍 

> Allowed passing of the product/service string (so I can set it to my service publicly, even though it is your code.)

I had no idea what you meant initially. -- Yeah that was in there because I originally developed this for... scheduling movie nights (it's a prepandemic thing.) I just extracted the code so I wasn't sure what to put there. I changed it to "arran4" in your commit -- although I feel I might regret that. 

> Allowed Serialization to an io.Writer not just string (better error handling here would be a nice addition, later.)
> Made an example RRule in the README 'cos I wasn't sure how to make that call (I assumed all sorts of KeyValue parameters.)

All good. Although I don't like the requirement to write Go code as well as ical so I will probably look at making that helper function a bit easier to use in the future (if I get an incentive to make major changes to this. There are a couple of changes I want to make for ease of use.) 

PR merged.

**Comment by arran4** [comment]
OH.. Didn't notice the linter complained:

>   Running [/home/runner/golangci-lint-1.35.2-linux-amd64/golangci-lint run --out-format=github-actions] in [] ...
>  Error: Error return value of `calendar.SerializeTo` is not checked (errcheck)

Hrm.

**Comment by arran4** [comment]
No idea how to handle that case without breaking either DRY or existing code.

**Comment by adamjack** [comment]
Hmm, sorry about that (and not replying sooner, I had tea verse keyboard ...) ... anyway, I kinda wanted error handle (since writing to a stream) but I didn't go far enough. It says this PR is closed, does that mean rejected? Should I work a little harder to inject error handling fuller or should I just patch this problem, e.g. some sort of _ =. To add an error return to your Serialize() will likely break usr code, so I assume you'd not want that.

**Comment by arran4** [comment]
Hey Adam.. No worries I don't expect immediate responses. The PR is closed because it was accepted!

No further work is necessary.. (I guess that makes my comments unnecessary? :P ) 

The last assumption is correct.

**Pull request merged**